src/components/logo.tsx

```diff
@@ -0,0 +1,16 @@
+import { Link } from '@tanstack/react-router'
+import { useIsMobile } from '@/lib/hooks/use-is-mobile'
+
+export function Logo({ to }: { to: string }) {
+  const isMobile = useIsMobile()
+
+  return (
+    <Link to={to}>
```

Comment:
nit: can we make this type safe?
```suggestion
import { Link, type LinkProps } from '@tanstack/react-router'
import { useIsMobile } from '@/lib/hooks/use-is-mobile'

type LogoProps = {
  linkProps: LinkProps
}

export function Logo({ linkProps }: LogoProps) {
  const isMobile = useIsMobile()

  return (
    <Link {...linkProps}>
```

Reply:
Great idea.